File: server/DTOs/AnalyticsDtos.cs
namespace FeelJournal.DTOs;

public class MoodAnalysisDto
{
    public List<EmotionFrequencyDto> EmotionFrequencies { get; set; } = new();
    public List<CategoryMoodDto> CategoryMoods { get; set; } = new();
    public Dictionary<string, int> EmotionByCategory { get; set; } = new();
}

public class EmotionFrequencyDto
{
    public string EmotionName { get; set; } = string.Empty;
    public string EmotionIcon { get; set; } = string.Empty;
    public int Count { get; set; }
    public double Percentage { get; set; }
}

public class CategoryMoodDto
{
    public string CategoryName { get; set; } = string.Empty;
    public string CategoryType { get; set; } = string.Empty;
    public string DominantEmotion { get; set; } = string.Empty;
    public string DominantEmotionIcon { get; set; } = string.Empty;
    public int DominantEmotionCount { get; set; }
    public int TotalEntries { get; set; }
}

public class ExportRequest
{
    public string Format { get; set; } = "CSV";
    public string? DateFrom { get; set; }
    public string? DateTo { get; set; }
}

File: app/src/main/java/com/example/feel_journal/data/remote/dto/AnalyticsDtos.kt
package com.example.feel_journal.data.remote.dto

import com.google.gson.annotations.SerializedName

data class MoodAnalysisDto(
    @SerializedName("emotionFrequencies") val emotionFrequencies: List<EmotionFrequencyDto>,
    @SerializedName("categoryMoods") val categoryMoods: List<CategoryMoodDto>,
    @SerializedName("emotionByCategory") val emotionByCategory: Map<String, Int>
)

data class EmotionFrequencyDto(
    @SerializedName("emotionName") val emotionName: String,
    @SerializedName("emotionIcon") val emotionIcon: String,
    @SerializedName("count") val count: Int,
    @SerializedName("percentage") val percentage: Double
)

data class CategoryMoodDto(
    @SerializedName("categoryName") val categoryName: String,
    @SerializedName("categoryType") val categoryType: String,
    @SerializedName("dominantEmotion") val dominantEmotion: String,
    @SerializedName("dominantEmotionIcon") val dominantEmotionIcon: String,
    @SerializedName("dominantEmotionCount") val dominantEmotionCount: Int,
    @SerializedName("totalEntries") val totalEntries: Int
)

data class ExportRequest(
    @SerializedName("format") val format: String = "CSV",
    @SerializedName("dateFrom") val dateFrom: String?,
    @SerializedName("dateTo") val dateTo: String?
)
